### PR TITLE
Add supplementary_file_flag to anvil_file

### PR DESCRIPTION
**Pull request by ncalvanese1**
Added new field and moved schema to version 5

### CODE REVIEW

File: anvil_schema/schema/mapping_schema_object.json [hunk 1]
```diff
@@ -147,6 +147,7 @@
         {"name": "reference_assembly", "datatype": "string", "array_of": true, "required": false},
         {"name": "file_name", "datatype": "string", "array_of": false, "required": false},
         {"name": "file_ref", "datatype": "fileref", "array_of": false, "required": false},
+        {"name": "supplementary_file_flag", "datatype": "boolean", "array_of": false, "required": false},
```
Comment (hannes-ucsc) [inline anvil_schema/schema/mapping_schema_object.json]:
```suggestion
        {"name": "is_supplementary", "datatype": "boolean", "array_of": false, "required": false},
```

Comment (hannes-ucsc) [inline anvil_schema/schema/mapping_schema_object.json]:
```suggestion
        {"name": "supplementary_file_flag", "datatype": "boolean", "array_of": false, "required": true},
```

What would `null` mean if we allowed the column to be nullable?

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
The column name change is a suggestion only. The nullability issue is what made me request changes for this.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
https://github.com/broadinstitute/anvil_tdr_ingest/pull/3#pullrequestreview-1307940150

**Comment by hannes-ucsc** [comment]
~I guess I don't have permission to request changes. If I do, GH changes the status to "suggested"~ Fixed now that I have accepted the invitation.

File: anvil_schema/schema/mapping_schema_object.json (hunk 1 above)
Comment (ncalvanese1) [reply anvil_schema/schema/mapping_schema_object.json]: Thanks for the quick review! The way we've written the logic to populate this field, it should never be null (but rather should always be either True or False). The only complication with making this field formally "required" is that we are populating it as a secondary process, and making the field required in TDR's purview would result in the initial load of the table failing unless we updated that code to pass a default value. From your standpoint, you can consider this field to be non-nullable.

Comment (hannes-ucsc) [reply anvil_schema/schema/mapping_schema_object.json]: Makes sense.

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**